### PR TITLE
HarfangLab EDR connector v1.2.0

### PR DESCRIPTION
**Pull request by Pierre-HarfangLab**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description

Version 1.2.0 of HarfangLab EDR connector with a lot of new commands to fetch new telemetry, manage whitelists, get process graphs...

## Must have
- [x] Tests (internal test suite not part of the repository)
- [x] Documentation

### CODE REVIEW

**Comment by ostolero** [comment]
Hi @Pierre-HarfangLab , thank you for this contribution!
I see you haven't filled out our contribution form yet, please fill that out so I can start reviewing this.
tnx

**Comment by ostolero** [comment]
Hi @Pierre-HarfangLab thank you for signing the contribution form!
Your PR review will be slightly delayed since it is the holiday season in Israel and we have a holiday this weekend (24th-25th of September), I apologize in advance.
In the meantime, you can also try to solve the errors in the build- there are some validation errors.
I'm available for any of your questions here or on Slack.
Thanks again.

**Review by ostolero — COMMENTED** [review]
Hi @Pierre-HarfangLab , thank you for the contribution!
The code looks good overall, I've added a few questions to clear some of the changes.
I've also reached through email to schedule a meeting for us to go over the problematic files that fail to upload to the server.
When we fix that we'll have another review by our security team for the changes in classifier,playbooks and incident fields.
Thanks and let me know if you need any assistance,

File: Packs/HarfangLabEDR/Integrations/Hurukai/Hurukai.py [hunk 1]
```diff
@@ -322,7 +517,7 @@ def search_ioc(self, ioc_value, source_id):
 
         return self._http_request(
             method='GET',
-            url_suffix='/api/data/threat_intelligence/IOCIndicator/',
+            url_suffix='/api/data/threat_intelligence/IOCRule/',
```
[the remark this answers is not in the document]
Comment (Pierre-HarfangLab) [reply Packs/HarfangLabEDR/Integrations/Hurukai/Hurukai.py:520]: Yes it is.

File: Packs/HarfangLabEDR/Integrations/Hurukai/Hurukai.py [hunk 2]
```diff
@@ -2324,8 +3275,10 @@ def main():
             args['alert_type'] = demisto.params().get('alert_type', None)
             args['min_severity'] = demisto.params().get(
                 'min_severity', SEVERITIES[0])
-            args['max_fetch'] = demisto.params().get('max_fetch', None)
-        target_function(client, args)
+            args['max_results'] = demisto.params().get('max_results', None)
```
[the remark this answers is not in the document]
Comment (Pierre-HarfangLab) [reply Packs/HarfangLabEDR/Integrations/Hurukai/Hurukai.py]: I looked at other integrations and then replaced max_results with "max_fetch" that should be more consistent.

File: Packs/HarfangLabEDR/ReleaseNotes/1_2_0.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+##### Hurukai alert mapper
+
+The Hurukai alert mapper has been modified to handle both `Hurukai alerts` and `Hurukai threats`. It has been extended with a couple of fields (`dbotMirrorDirection`, `dbotMirrorId`, `dbotMirrorInstance`) in order to manage mirroring of alert/threats' statuses between a HarfangLab EDR manager and an XSOAR. 
+
+#### Playbooks
+
+##### Hurukai - Alert management
+
+- %%UPDATE_RN%%
```
[the remark this answers is not in the document]
Comment (Pierre-HarfangLab) [reply Packs/HarfangLabEDR/ReleaseNotes/1_2_0.md]: Duplicate documentation section that has been removed...

File: Packs/HarfangLabEDR/Integrations/Hurukai/Hurukai.yml [hunk 4]
```diff
@@ -55,13 +54,52 @@ configuration:
...
+  - Incoming And Outgoing
+- defaultvalue: 'Security Events'
+  display: 'Fetch types'
+  name: fetch_types
+  options:
+  - 'Security Events'
+  - 'Threats'
+  type: 16
+  advanced: True
+
```
[the remark this answers is not in the document]
Comment (Pierre-HarfangLab) [reply Packs/HarfangLabEDR/Integrations/Hurukai/Hurukai.yml]: Done

**Comment by melamedbn** [comment]
Hello @Pierre-HarfangLab,

I hope this finds you well. I've taken a look at the content you've shared, and it's evident that a lot of thought and effort went into its creation! I have a few recommendations to enhance the content's structure and usability:

### Incident Fields:
1. Please prefix all incident fields with "Hurukai". This would mean changes in associated mappers, classifier, playbook, and layout. For instance:
   - `Top Rules` would become `Hurukai Top Rules`.

### Mappers & Classifiers:
1. For naming consistency, please follow the convention: 'product name - Incoming Mapper'. For example:
   - `Hurukai - Incoming Mapper`.
   - The same applies to outgoing mapper and classifier.
2. A demo showcasing how the outgoing mapper functions would be incredibly helpful.

### Incident Type:
1. Should the `Hurukai Alert Management` playbook be the default for this incident type?

### Playbook: Hurukai - Alert Management:
1. I noticed there aren't any inputs or outputs defined for the playbook. Is this intentional?
2. Some task titles could be made more concise, improving readability:
   - Task 64: **'Should I isolate the endpoint before starting the investigation?'** can be shortened to **'Should isolate before investigating?'**.
3. To streamline, consider removing auxiliary details from task titles and move them to the task description:
   - For Task 17, **'Get endpoint information (OS, policy...)'**, perhaps just **'Get endpoint information'** with the specifics in the description.
4. For Task 19, it might be worth considering if we want to recover the endpoint from isolation without seeking user confirmation.
5. It's crucial to note that the forensics playbooks lack outputs, which means analysts won't see results. Perhaps it's worth adding outputs to them to present any vital information aiding the analyst's decisions.

Your dedication is clear in this content, and these suggestions aim to polish and refine it further. If any clarification is required or you'd like to discuss any point, please don't hesitate to get in touch.

Warm regards,

Ben

**Comment by omerKarkKatz** [comment]
Hi @Pierre-HarfangLab, we haven’t heard from you in a while. 
Do you need any help with the pull request?
Please feel free to reach out here or on Slack.
Thanks again for contributing to our repo, hope to hear from you soon.

**Comment by omerKarkKatz** [comment]
@Pierre-HarfangLab, As discussed by email this PR will be closed for now, Please reopen it once you are ready to continue working on the contribution. 
Thanks a lot!

**Comment by omerKarkKatz** [comment]
@Pierre-HarfangLab, As discussed by email this PR will be closed for now, Please reopen it once you are ready to continue working on the contribution. 
Thanks a lot!

**Pull request closed without merging**